syntax = "proto3";

package devices;

message GetDevicesRequest {
}

message CreateDeviceRequest {
    string name = 1;
    string mac_address = 2;
}

message UpdateDeviceRequest {
    string id = 1;
    string name = 2;
    string mac_address = 3;
}

message DeleteDeviceRequest {
    string id = 1;
}

message Device {
    string id = 1;
    string name = 2;
    string mac_address = 3;
    bool is_active = 4;
    string last_read = 5;
    string created_at = 6;
    string updated_at = 7;
}

message GetDevicesResponse {
    repeated Device devices = 1;
}

message CreateDeviceResponse {
    Device device = 1;
}

message UpdateDeviceResponse {
    Device device = 1;
}

message DeleteDeviceResponse {
    string message = 1;
}

service DeviceService {
    rpc GetAllDevices(GetDevicesRequest) returns (GetDevicesResponse);
    rpc CreateDevice(CreateDeviceRequest) returns (CreateDeviceResponse);
    rpc UpdateDevice(UpdateDeviceRequest) returns (UpdateDeviceResponse);
    rpc DeleteDevice(DeleteDeviceRequest) returns (DeleteDeviceResponse);
}
